naptime/src/main/scala/org/coursera/naptime/access/AccessControl.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 Coursera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.coursera.naptime.access
+
+import org.coursera.naptime.NaptimeActionException
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+/**
+ * Control access to a resource method. Arbitrary, asynchronous logic may be executed, but
+ * successful access grant must produce an authentication object of type [[A]].
+ *
+ * @tparam R request data (for example, for Play HTTP requests, this is
```

Comment:
@saeta: With just this PR, it's not useful, because it's not used anywhere. My longer-term purpose is to continue pushing Naptime towards not being completely tied to HTTP, and changing access control to not exclusively be about `RequestHeader` is part of that. I'm imagining that the `Rest.auth(...)` DSL will be expanded to handle general `AccessControl` rather than just `HeaderAccessControl` (which I think will have more consequences than just this PR on how access controls are implemented).